integration_tests/src/main/python/data_gen.py

```diff
@@ -681,6 +730,19 @@ def skip_if_not_utc():
     if (not is_tz_utc()):
         skip_unless_precommit_tests('The java system time zone is not set to UTC')
 
+# SparkSession with the same runtime configs will have same memory address and vice versa, 
```

Comment:
I recorded the max memory usage of a single test of all current IT, it's 7282688 bytes. so it may cost additional $128*7282688$ bytes $= 932$ MB memory if every slots with huge data sets. Not very large to cause an OOM, is it?

Reply:
sounds good to me,  can you also help add this numbers in comment for reference? so if we met any related issue people could try poke the cache size to quick workaround, thanks

Reply:
Ok, I added it in description.

Reply:
I mean add those in this change, maybe as a **NOTE** comment

Reply:
Ok, added a comment here